p2p/stdnetwork/network.go

```diff
@@ -19,45 +22,49 @@ var (
 )
 
 type network struct {
+	host   libp2pHost.Host
 	pstore sharedP2P.Peerstore
 
 	logger *modules.Logger
 }
 
-func NewNetwork(bus modules.Bus, pstoreProvider providers.PeerstoreProvider, currentHeightProvider providers.CurrentHeightProvider) (n typesP2P.Network) {
+func NewNetwork(host libp2pHost.Host, pstoreProvider providers.PeerstoreProvider, currentHeightProvider providers.CurrentHeightProvider) (typesP2P.Network, error) {
 	networkLogger := logger.Global.CreateLoggerForModule("network")
 	networkLogger.Info().Msg("Initializing stdnetwork")
 
 	pstore, err := pstoreProvider.GetStakedPeerstoreAtHeight(currentHeightProvider.CurrentHeight())
 	if err != nil {
-		networkLogger.Fatal().Err(err).Msg("Error getting peerstore")
+		return nil, err
 	}
 
 	return &network{
+		host:   host,
 		logger: networkLogger,
 		pstore: pstore,
-	}
+	}, nil
 }
 
-// TODO(olshansky): How do we avoid self-broadcasts given that `AddrBook` may contain self in the current p2p implementation?
 func (n *network) NetworkBroadcast(data []byte) error {
 	for _, peer := range n.pstore.GetPeerList() {
-		if _, err := peer.GetStream().Write(data); err != nil {
-			n.logger.Error().Err(err).Msg("Error writing to one of the peers during broadcast")
+		if err := utils.Libp2pSendToPeer(n.host, data, peer); err != nil {
+			n.logger.Error().
+				Err(err).
+				Bool("TODO", true).
+				Str("pokt address", peer.GetAddress().String()).
+				Msg("broadcasting to peer")
```

Comment:
It seems to me like the distinction you're making between "broadcast" and "send" is a matter of routing given message to a given node. Where "broadcasting" is an indirect send to the target node. I.e.: it will also "send" but not directly to the target, the message will be propagated towards the target by other node(s) which receive copies of same message.

I don't think it makes sense to think of "broadcast" as targeting any specific node; but rather to think of its target as the network (or network segment, classically). As we don't currently have a conception of a "broadcast address" (nor do we need one right now), the "network address" degree of freedom reduces to zero. Subsequently, when I read this, I think something like "this is the peer this node is currently sending to in order to broadcast the given message". Does that make sense? Do you have any suggestions on how to better reflect that in the code, comments, and/or docs?